src/npcmove.cpp

```diff
@@ -1117,11 +1117,33 @@ void npc::move_to(int x, int y)
             g->m.creature_in_field( *this );
         } else if (g->m.open_door(x, y, !g->m.is_outside( posx(), posy() ) ) ) {
             moves -= 100;
-        } else if (g->m.is_bashable(x, y) && g->m.bash_rating(str_cur + weapon.type->melee_dam, x, y) > 0) {
-            moves -= int(weapon.is_null() ? 80 : weapon.attack_time() * 0.8);;
-            int smashskill = str_cur + weapon.type->melee_dam;
-            g->m.bash( x, y, smashskill );
+        } else if (not g->m.has_flag_ter_or_furn( "ELECTRIFIED", x, y )) {
+        bool ter_or_furn = g->m.has_flag_ter_or_furn( "CLIMBABLE", x, y );
+            if (ter_or_furn) {
+            bool u_see_me = g->u.sees( *this );
+            int climb = dex_cur;
+                if (one_in( climb )) {
+                    if (u_see_me == true) {
```

Comment:
Why do people compare a boolean to another boolean. It is already a boolean, comparing it to true is pointless. Just write `if(u_see_me)`